### ISSUE TITLE
Inherit on modules ignores type

### ISSUE THREAD

**Issue by nathanchere** [jaagr/polybar#1339]
As per title. When inheriting configuration from another module, the `type` field doesn't appear to be inherited.

Example of what I think should work but doesn't:

```
[module/div-base]
type = custom/text
content-padding = 0
content-background = ${colors.black}
content-foreground = ${colors.white}

[module/div-right]
inherit = module/div-base
content = <

[module/div-left]
inherit = module/div-base
content = >
```

Actual result: 

> error: Disabling module "div-right" (reason: Missing parameter "module/div-right.type")
> error: Disabling module "div-left" (reason: Missing parameter "module/div-left.type")

However it works if I do this:

```
[module/div-base]
content-padding = 0
content-background = ${colors.black}
content-foreground = ${colors.white}

[module/div-right]
inherit = module/div-base
type = custom/text
content = <

[module/div-left]
inherit = module/div-base
type = custom/text
content = >
```

I don't know if this is still an issue with the new parser outlined in https://github.com/jaagr/polybar/pull/1237

**Comment by patrick96** [comment]
The new parser should not have that problem, but the current one shouldn't either. I don't see anything in the code that would suggest it excludes the `type` key. I'm also not able to reproduce this with the following config:

```
[colors]
black = #111111
white = #eeeeee

[bar/example]
width = 100%
height = 27

modules-left = div-right div-left

[module/div-base]
type = custom/text
content-padding = 0
content-background = ${colors.black}
content-foreground = ${colors.white}

[module/div-right]
inherit = module/div-base
content = <

[module/div-left]
inherit = module/div-base
content = >
```

Could you please post the entire config you used which didn't work, as well as the output of `polybar -vvv`

**Comment by nathanchere** [comment]
Simplified example for clarity, but it happens with this too:
```
[bar/top]
bottom = false
width = 90%
height = 25
modules-right  =  div-r1 

[module/div-base]
type = custom/text
content-padding = 0

[module/div-right]
inherit = module/div-base
content = 

[module/div-r1]
inherit = module/div-right
content-background = #BF0
content-foreground = #000
```

Result: 
> error: Disabling module "div-r1" (reason: Missing parameter "module/div-r1.type")

If I add the type on `div-right` it works, i.e. 

```
[module/div-base]
type = custom/text
content-padding = 0

[module/div-right]
inherit = module/div-base
type = custom/text
content = 

[module/div-r1]
inherit = module/div-right
content-background = #BF0
content-foreground = #000
```

so appears to be related to nested inherits, not the 'type' field specifically. I confirmed by moving other field up the chain and seeing them not passed down. I found another issue here mentioning nested inherits but the solution was cited as being alphabetically ordered loading of modules. I thought mine fit that (base < right), but if I rename the modules like as follows it works too:

```
[module/div-base1]
type = custom/text
content-padding = 0

[module/div-base2]
type = custom/text
content = 

[module/div-r1]
inherit = module/div-base2
```

`polybar -vvv`: 
```
polybar 3.1.0-161-g028b141

Features: +alsa +curl +i3 -mpd +network +pulseaudio +xkeyboard

X extensions: +randr (+monitors) -render -damage -sync -composite +xkb +xrm +xcursor

Build type: Debug
Compiler: /usr/bin/c++
Compiler flags: -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -fno-plt -Wall -Wextra -Werror -Wno-noexcept-type -O2 -pedantic -pedantic-errors
Linker flags: -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now
```

**Comment by patrick96** [comment]
This would make this a duplicate of #412. The new config parser certainly fixes this.

The specific problem here is that `div-r1 < div-right` thus first all the keys from `div-right` get copied into `div-r1` (this includes the `inherit = module/div-base` key),only after that is the inherit directive in `div-right` applied. In the end `div-r1` has the `inherit = module/div-base` but it never got applied, so it misses the keys from `div-base`. This happens because the inherit directives are only applied once and not recursively.
